apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,32 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
-            ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            if (_clientTask.IsCompleted)
+            {
+                ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                return _clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            }
+
+            if (Interlocked.Increment(ref _taskCounter) <= MaxWaitingTasks)
+            {
+                return _clientTask.ContinueWith(clientTask =>
+                {
+                    if (clientTask.IsFaulted)
```

Comment:
If we're happy throwing an exception, just using `clientTask.Result` would do the right thing. Maybe just add a comment to make it clear that it's intentional?